protocol-units/settlement/mcr/client/src/eth_client.rs

```diff
@@ -0,0 +1,536 @@
+		});
+		Ok(Box::pin(stream) as CommitmentStream)
+	}
+
+	async fn get_commitment_at_height(
+		&self,
+		height: u64,
+	) -> Result<Option<BlockCommitment>, anyhow::Error> {
+		let contract = MCR::new(self.config.mrc_contract_address.parse()?, &self.ws_provider);
+		let MCR::getValidatorCommitmentAtBlockHeightReturn { _0: commitment } = contract
+			.getValidatorCommitmentAtBlockHeight(U256::from(height), self.signer_address)
+			.call()
+			.await?;
+		let return_height: u64 = commitment.height.try_into()?;
+		// Commitment with height 0 mean not found
+		Ok((return_height != 0).then_some(BlockCommitment {
+			height: commitment.height.try_into()?,
+			block_id: Id(commitment.blockId.into()),
+			commitment: Commitment(commitment.commitment.into()),
+		}))
+	}
+
+	async fn get_max_tolerable_block_height(&self) -> Result<u64, anyhow::Error> {
+		let contract = MCR::new(self.config.mrc_contract_address.parse()?, &self.ws_provider);
+		let MCR::getMaxTolerableBlockHeightReturn { _0: block_height } =
+			contract.getMaxTolerableBlockHeight().call().await?;
+		let return_height: u64 = block_height.try_into()?;
+		Ok(return_height)
+	}
+}
+
+#[cfg(test)]
+pub mod test {
+	use super::*;
+	use alloy_provider::ProviderBuilder;
+	use alloy_signer_wallet::LocalWallet;
+	use movement_types::Commitment;
+	use std::env;
+
+	// Define 2 validators (signer1 and signer2) with each a little more than 50% of stake.
+	// After genesis ceremonial, 2 validator send the commitment for height 1.
+	// Validator2 send a commitment for height 2 to trigger next epoch and fire event.
+	// Wait the commitment accepted event.
+	//#[ignore]
+	#[tokio::test]
+	async fn test_send_commitment() -> Result<(), anyhow::Error> {
+		//Activate to debug the test.
+		// use tracing_subscriber::EnvFilter;
+
+		// tracing_subscriber::fmt()
+		// 	.with_env_filter(
+		// 		EnvFilter::try_from_default_env().unwrap_or_else(|_| EnvFilter::new("info")),
+		// 	)
+		// 	.init();
+
+		// Inititalize Test variables
+		let rpc_port = env::var("MCR_ANVIL_PORT").unwrap();
+		let rpc_url = format!("http://localhost:{rpc_port}");
+		let ws_url = format!("ws://localhost:{rpc_port}");
+
+		let anvil_address = read_anvil_json_file_address()?;
+
+		//Do SC ceremony init stake calls.
+		do_genesis_ceremonial(&anvil_address, &rpc_url).await?;
+
+		let mcr_address = read_mcr_sc_adress()?;
+		//Define Signers. Ceremony define 2 signers with half stake each.
+		let signer1: LocalWallet = anvil_address[1].1.parse()?;
+		let signer1_addr = signer1.address();
+
+		//Build client 1 and send first commitment.
+		let provider_client1 = ProviderBuilder::new()
+			.with_recommended_fillers()
+			.signer(EthereumSigner::from(signer1))
+			.on_http(rpc_url.parse().unwrap());
+
+		let config = McrEthSettlementConfig {
+			mrc_contract_address: mcr_address.to_string(),
+			gas_limit: DEFAULT_TX_GAS_LIMIT,
+			tx_send_nb_retry: MAX_TX_SEND_RETRY,
+		};
+
+		let client1 = McrEthSettlementClient::build_with_provider(
```

Comment:
Yeah, so eventually, what I'll want to have is a `try_from_env` constructor defined that would wrap this all up. But, for now, this is good.

Reply:
I've implemented one version.